extensions/sql-migration/src/api/sqlUtils.ts

```diff
@@ -509,3 +510,73 @@ export async function isSourceConnectionSysAdmin(): Promise<boolean> {
 
 	return getSqlBoolean(results.rows[0][0]);
 }
+
+export async function canTargetConnectToStorageAccount(
+	targetType: MigrationTargetType,
+	targetServer: SqlManagedInstance | SqlVMServer | AzureSqlDatabaseServer,
+	storageAccount: StorageAccount,
+	account: azdata.Account,
+	subscription: Subscription): Promise<boolean> {
+
+	// additional ARM properties of storage accounts which aren't exposed in azurecore
+	interface StorageAccountAdditionalProperties {
+		publicNetworkAccess: string,
+		networkAcls: NetworkRuleSet,
+		privateEndpointConnections: PrivateEndpointConnection[]
+	}
+	interface NetworkRuleSet {
+		virtualNetworkRules: VirtualNetworkRule[],
+		defaultAction: string
+	}
+	interface VirtualNetworkRule {
+		id: string,
+		state: string,
+		action: string
+	}
+
+	const storageAccountProperties: StorageAccountAdditionalProperties = (storageAccount as any)['properties'];
+	const storageAccountPublicAccessEnabled: boolean = storageAccountProperties.publicNetworkAccess ? storageAccountProperties.publicNetworkAccess.toLowerCase() === 'Enabled'.toLowerCase() : true;
+	const storageAccountDefaultIsAllow: boolean = storageAccountProperties.networkAcls ? storageAccountProperties.networkAcls.defaultAction.toLowerCase() === 'Allow'.toLowerCase() : true;
```

Comment:
Are defaultAction and virtualNetworkRules guarantee they will never be undefined?

Reply:
added a check